compiler/rustc_passes/src/diagnostic_items.rs

```diff
@@ -37,6 +37,8 @@ impl<'v, 'tcx> ItemLikeVisitor<'v> for DiagnosticItemCollector<'tcx> {
     fn visit_impl_item(&mut self, impl_item: &hir::ImplItem<'_>) {
         self.observe_item(&impl_item.attrs, impl_item.hir_id);
     }
+
+    fn visit_foreign_item(&mut self, _foreign_item: &hir::ForeignItem<'_>) {}
```

Comment:
```suggestion
    fn visit_foreign_item(&mut self, foreign_item: &hir::ForeignItem<'_>) {
        self.observe_item(foreign_item.attrs, foreign_item.hir_id);
    }
```

we currently do this manually in line 107, but should just do this here.